### PR TITLE
[exporter/datadog] Replace HistogramMode defined as string with enum

### PR DESCRIPTION
**Pull request by keisku**
**Description:** <Describe what has changed.>

Replace [HistogramConfig.Mode](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/033bc949bec36233c6c4a050a951f119dd00b16d/exporter/datadogexporter/config/config.go#L105) defined as string with [translator.HistogramMode](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/033bc949bec36233c6c4a050a951f119dd00b16d/exporter/datadogexporter/internal/model/translator/config.go#L89).

**Link to tracking Issue:** <Issue number if applicable> https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8373

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: CHANGELOG.md
Comment (mx-psi) [inline CHANGELOG.md:54]: This is technically a breaking change for the Go API, since it will make some code fail to compile: https://go.dev/play/p/KtWLxjhP6az I think it is fine to do this on a single release since it's hard to do over several releases, and the impact is minimal. However, just in case, let's move this to the "🛑 Breaking changes 🛑" section instead of Enhancements.

**Review by mx-psi — COMMENTED** [review]
Code LGTM, just one comment about the changelog. For reference, this relates to #8373

File: CHANGELOG.md
Comment (keisku) [reply CHANGELOG.md]: @mx-psi Thanks for your review! Moved to Breaking changes section. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9589/commits/c049c656ca153cc5378add68e3351d38b7844e0e

File: exporter/datadogexporter/config/config.go
Comment (mx-psi) [inline exporter/datadogexporter/config/config.go:519]: One last thing (sorry just saw this now): this switch can be removed since `UnmarshalText` will check for this

**Review by mx-psi — APPROVED** [review]
LGTM, I left a comment for one last 'nice to have' thing I forgot on my earlier review

File: exporter/datadogexporter/config/config.go
Comment (keisku) [reply exporter/datadogexporter/config/config.go:519]: @mx-psi 
Should I remove the following code?
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c049c656ca153cc5378add68e3351d38b7844e0e/exporter/datadogexporter/config/config.go#L518-L523

Comment (mx-psi) [reply exporter/datadogexporter/config/config.go:519]: Yes, that's it

Comment (keisku) [reply exporter/datadogexporter/config/config.go]: Removed it! https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9589/commits/5eb122982d1cbfc09c17afa54a488051376199ef

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
For a maintainer to double check before merging: this has a breaking change on the Go API, since we switch from a string to string type alias. It will not break users of the binary. I believe the breakage is okay since it's hard to do in two steps and disruption is minimal, but I want a maintainer to explicitly approve this since it goes against the guidelines.

**Review by codeboten — APPROVED** [review]
I think this is ok so long as you as the code owner for this component are ok with this change 👍

**Pull request merged**
